Test follow_dot_cursor

el0qu4nce opened the pull request:
pull request with updated follow_dot_cursor

el0qu4nce committed:
Update graphs_creator.py

adapting creator for cursor

el0qu4nce committed:
Update graphs_creator.py

getting back

el0qu4nce committed:
Update graphs_creator.py

el0qu4nce committed:
Create follow_dot_cursor.py

updated follow_dot_cursor v 1.0.2

A1t3r commented on follow_dot_cursor.py:
```diff
@@ -0,0 +1,94 @@
+
+class FollowDotCursor(object):
+
+    def __init__(self, ax, x, y, tolerance=5, formatter=fmt, offsets=(-20, 20)):
```
вижу много лишних полей с инита, которые никак не используются, убери плз

A1t3r commented on follow_dot_cursor.py:
```diff
@@ -0,0 +1,94 @@
+
+
+def fmt(x, y):
+    csv_file = csv.reader(open('data/processed_weather_data.csv', "r"),
```
на текущий момент мы можем добавлять в pandas dataframe данные, ты же перечитываешь файл эти данные не учитывая. Я предлагаю использовать именно тот датафрейм, который хранит DataStorage в поле self._df. У пандаса есть возможность выполнять запросы на поиск данных, в твоём случае это может выглядеть так:
```
def fmt(x, y, df):
    return f'{df[(df["Humidity"]==y) & (df["Temperature (C)"]==x)]["Formatted Date"]}'.format(x=x, y=y)
```
Но лучшим решением было бы организовать оси в трехмерном виде, в данный момент у нас есть разделение осей по температуре и влажности, можно добавить третью ось времени, но не отображать её для графика в main_window, эта ось как бы смотрит на нас и становится нам видна только при наведении мышки

el0qu4nce committed:
Update follow_dot_cursor.py

updated date-search by specs via pandas dataframe structure. 
also deleted not-used init val-s.
